assets/app/view/game/part/track_node_path.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require 'view/game/part/base'
+
+module View
+  module Game
+    module Part
+      class TrackNodePath < Base
+        needs :tile
+        needs :path
+        needs :color, default: 'black'
+        needs :width, default: 8
+        needs :dash, default: '0'
+
+        EDGE_PERP_ANGLES = [90, 30, -30, -90, -150, 150].freeze
+
+        EDGE_REGIONS = {
+          0 => TRACK_TO_EDGE_0,
+          1 => TRACK_TO_EDGE_1,
+          2 => TRACK_TO_EDGE_2,
+          3 => TRACK_TO_EDGE_3,
+          4 => TRACK_TO_EDGE_4,
+          5 => TRACK_TO_EDGE_5,
+        }.freeze
+
+        def load_from_tile
+          # for now assumes one edge and one node on path
+          @edge = @path.edges.first.num
+          @terminal = @path.terminal
+          @junction = @path.junction
+          return unless @path.stop
+
+          @stop = @path.stop
+          @ct_edge = @tile.preferred_city_town_edges[@stop]
+        end
+
+        def begin_location
+          {
+            x: -Math.sin((@edge * 60) / 180 * Math::PI) * 87,
+            y: Math.cos((@edge * 60) / 180 * Math::PI) * 87,
+          }
+        end
+
+        def end_location
+          return { x: 0, y: 0 } if @junction
+
+          if (@tile.towns.size + @tile.cities.size) < 2
```

Comment:
could this be @tile.nodes.size < 2 ? nodes includes offboards, cities, junctions, and towns

Reply:
Yes, won't be called with offboards and junctions are handled above.
Might be better as @tile.stops < 2

Reply:
cool, stops makes sense

Reply:
And now I know why it was written this way. @tile.stops and @tile.nodes doesn't find cities/towns with no paths connected to them :-/